core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/HttpGoogleServicesDAO.scala

```diff
@@ -423,18 +423,24 @@ class HttpGoogleServicesDAO(
     }
   }
 
-  override def getBucket(bucketName: String)(implicit executionContext: ExecutionContext): Future[Option[Bucket]] = {
+  override def getBucket(bucketName: String, userProject: Option[GoogleProjectId])(implicit executionContext: ExecutionContext): Future[Option[Bucket]] = {
     implicit val service = GoogleInstrumentedService.Storage
     retryWithRecoverWhen500orGoogleError(() => {
       val getter = getStorage(getBucketServiceAccountCredential).buckets().get(bucketName)
-      Option(executeGoogleRequest(getter))
+
+      val getterWithProject = userProject match {
+        case Some(project) => getter.setUserProject(project.value)
+        case None => getter
+      }
```

Comment:
optional, for style: the `setUserProject()` method mutates the `Get`  object in place, Java-style, instead of returning a new copy in typical Scala style. Thus, you don't need to care about its return value, and you could write this terser:
```
val getter = getStorage(getBucketServiceAccountCredential).buckets().get(bucketName)
userProject.map( p => getter.setUserProject(p.value) )
Option(executeGoogleRequest(getter))
```

of course,  please test this and make sure it works, if you adopt it!

(this applies to a few places in this file)